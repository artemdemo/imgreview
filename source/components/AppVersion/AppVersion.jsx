import React from 'react';

import './AppVersion.less';

class AppVersion extends React.PureComponent {
    constructor(props) {
        super(props);

        this.state = {
            version: 'x.xx',
        };
    }

    componentDidMount() {
        const appVersionEl = document.querySelector('[name="app-version"]');
        if (appVersionEl) {
            this.setState({
                version: appVersionEl.getAttribute('content'),
            });
        }
    }

    render() {
        return (
            <div className='app-version'>
                {this.state.version}
            </div>
        );
    }
}

export default AppVersion;
