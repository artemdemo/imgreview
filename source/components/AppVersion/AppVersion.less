.app-version {
    position: fixed;
    bottom: 0;
    right: 5px;
    opacity: 0.1;
}
