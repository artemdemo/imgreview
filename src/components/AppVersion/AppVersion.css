.AppVersion {
  position: fixed;
  bottom: 0;
  right: 5px;
  opacity: 0.2;
}
