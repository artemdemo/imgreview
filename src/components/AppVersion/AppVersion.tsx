import React from 'react';
import styled from 'styled-components';

const AppVersionSty = styled.div`
    position: fixed;
    bottom: 0;
    right: 5px;
    opacity: 0.2;
`;

type TProps = {};

type TState = {
    version: string,
};

class AppVersion extends React.PureComponent<TProps, TState> {
    private readonly defaultVersion = 'x.xx';

    state = {
        version: this.defaultVersion,
    };

    componentDidMount() {
        const appVersionEl = document.querySelector('[name="app-version"]');
        if (appVersionEl) {
            this.setState({
                version: appVersionEl.getAttribute('content') || this.defaultVersion,
            });
        }
    }

    render() {
        return (
            <AppVersionSty>
                {this.state.version}
            </AppVersionSty>
        );
    }
}

export default AppVersion;
