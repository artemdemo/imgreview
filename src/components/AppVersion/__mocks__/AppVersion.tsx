import React from 'react';

const AppVersion = (props) => (
  <div data-mock="AppVersion">{props.children}</div>
);

export default AppVersion;
