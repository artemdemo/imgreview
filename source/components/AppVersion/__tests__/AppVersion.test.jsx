import React from 'react';
import renderer from 'react-test-renderer';
import AppVersion from '../AppVersion';

describe('AppVersion', () => {
    it('should render', () => {
        const tree = renderer.create(
            <AppVersion />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with version', () => {
        const querySelectorOrigin = document.querySelector;
        const version = '1.00.00-mock';
        const getAttributeMock = jest.fn(() => {
            return version;
        });
        const querySelectorMock = jest.fn(() => ({
            getAttribute: getAttributeMock,
        }));
        document.querySelector = querySelectorMock;
        const tree = renderer.create(
            <AppVersion />
        ).toJSON();

        expect(tree).toMatchSnapshot();
        expect(querySelectorMock).toBeCalledWith('[name="app-version"]');
        expect(getAttributeMock).toBeCalledWith('content');
        document.querySelector = querySelectorOrigin;
    });
});
