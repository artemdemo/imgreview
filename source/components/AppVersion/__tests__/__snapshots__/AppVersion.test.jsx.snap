// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppVersion should render 1`] = `
<div
  className="app-version"
>
  x.xx
</div>
`;

exports[`AppVersion should render with version 1`] = `
<div
  className="app-version"
>
  1.00.00-mock
</div>
`;
